File: 0055-jump-game/0055-jump-game.java
class Solution {

    public boolean canJump(int[] nums) {
        if (nums.length == 1) return true;
        
        boolean[] arr = new boolean[nums.length];
        arr[0] = true;
        for (int i = 0; i < nums[0] && i + 1 < nums.length; i++) {
            arr[i + 1] = true;
        }
        
        for (int i = 1; i < nums.length; i++) {
            if (!arr[i]) return false;
            for (int j = i; j < i + nums[i] && j + 1 < nums.length; j++) {
                arr[j + 1] = true;
            }
            
            if (arr[arr.length - 1]) return true;
        }
        
        return false;
    }

}

File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let reachable = 0;
    for (i = 0; i < nums.length; i++) {
        if (i > reachable) return false;
        reachable = Math.max(reachable, i + nums[i]);
        if (reachable >= nums.length - 1) break;
    }
    
    return true;
};